File: monitoreo/src/main/java/models/entidades/persona/autentificacion/Usuario.java
package models.entidades.persona.autentificacion;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDate;

@Entity
@Table(name="usuario")
public class Usuario {

  @Getter @Setter
  @Column(name = "id")
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private int id;

  @Getter @Setter
  @Column(name = "usuario")
  private String usuario;

  @Setter @Getter
  @Column(name = "contrasenia")
  private String contrasenia;

  @Getter@Setter
  @Column(name = "rol")
  @Enumerated(EnumType.STRING)
  private Rol rol;

  @Setter @Getter
  @Column(name="fecha_contraseña",columnDefinition = "date")
  private LocalDate fechaContrasenia;

  @Setter @Getter
  @Column(name = "mail")
  private String correoElectronico;

  public Usuario(String contrasenia, String correoElectronico)
  {
    this.setCorreoElectronico(correoElectronico);
    this.setContrasenia(contrasenia);
    this.setFechaContrasenia(LocalDate.now());
  }

  public Usuario(){}
}

File: MonitoreoCNET/MonitoreoCNET/Models/Persona/Usuario.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MonitoreoCNET.Models.Persona
{
    public enum Rol
    {
        MIEMBRO,
        ADMIN_COMUNIDAD,
        ADMIN_PLATAFORMA,
        RESPONSABLE
    }
    
    [Table("usuario")]
    public class Usuario
    {
        [Key]
        [Column("id")]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [Column("usuario")]
        public string UserName { get; set; }

        [Column("contrasenia")]
        public string Password { get; set; }

        [Column("rol")]
        public Rol Rol { get; set; }

        [Column("fecha_contraseña", TypeName = "date")]
        public DateTime FechaContrasenia { get; set; }

        [Column("mail")]
        public string CorreoElectronico { get; set; }

        public Usuario(string contrasenia, string usuario)
        {
            UserName = usuario;
            Password = contrasenia;
            FechaContrasenia = DateTime.Now;
        }

        private Usuario()
        {
        }
    }
}